### DISCUSSION TITLE
Redis Streams Implementation

### DISCUSSION THREAD

**Discussion by SSaiPrasad7** [electric-sheep-co/arduino-redis#38] [Q&A] [answered]
Hello arduino-redis team,
I would like to implement redis streams by using "arduino-redis"client library. Can I do that?
If possible, Can you please send the sample code?

**Comment by rpj** [comment, marked as answer]
Hi @saiprasadsamudrala,

Thank you for reaching out! Streams support could _definitely_ be added to this library, it's just something I haven't had the time to get to myself yet. I would _love_ it if you put together an implementation, just submit it as a pull request and I'll happily review and merge it!

As for implementation details, it's (hopefully) pretty straightforward: all the commands are implemented in [`Redis.cpp`](https://github.com/electric-sheep-co/arduino-redis/blob/master/Redis.cpp) and you'll note there are a couple macros defined to reduce the code duplication each requires. Additionally, please be sure to add a test for each added command in [`test/test.ino`](https://github.com/electric-sheep-co/arduino-redis/blob/master/test/test.ino).

Feel free to reach out with any further questions, I'm more than happy to help in any way I can. Thank you again for your interest and willingness to help, it is very much appreciated! 🙌

**Reply by rpj** [reply]
I've gone ahead and created an issue (#39) from this discussion in order to properly track the Streams implementation. 🚀
